agent/http.go

```diff
@@ -257,8 +257,17 @@ func (s *HTTPServer) handler(enableDebug bool) http.Handler {
 			metrics.MeasureSince(key, start)
 		}
 
-		gzipWrapper, _ := gziphandler.GzipHandlerWithOpts(gziphandler.MinSize(0))
-		gzipHandler := gzipWrapper(http.HandlerFunc(wrapper))
+		var gzipHandler http.Handler
+		minSize := gziphandler.DefaultMinSize
+		if pattern == "/v1/agent/monitor" {
+			minSize = 0
+		}
```

Comment:
Given that we want to continue supporting the monitor endpoint being gzipped, and that is the only endpoint that diverges from using the `DefaultMinSize`, I think we should add some test to the `agent` package that ensures `Accept-Encoding: gzip` is valid for this endpoint.

I know that the `command/monitor` package tests end up failing if you don't do this, but the failure is not obvious. I think it would be pretty easy to add a subtest in `TestAgent_Monitor` that adds the `Accept-Encoding` header to the request and has a good failure message.